### PR TITLE
Adding ingress(adj) to the SSG

### PR DESCRIPTION
**Pull request by skrthomas**
https://github.com/redhat-documentation/supplementary-style-guide/issues/366

### CODE REVIEW

**Comment by skrthomas** [comment]
@mportman12 here's my PR for #366

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: @bergerhoffer I added markup for mentionings of the `Ingress` resource. I also added some context to the *Incorrect forms* description. Wdyt? Does this clarify the ambiguity a little further? Any other things that are a little unclear still? 🤔

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:337]:
```suggestion
*Description*: _Ingress_ describes the incoming direction of network traffic. In Red Hat OpenShift, this traffic is often entering (ingress) or leaving (egress) an OCP cluster. The Ingress resource defines rules for how ingress network traffic should be routed from external sources into your services. An Ingress Controller is responsible for implementing these rules.
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:332]: This is confusing because the entry is for "Ingress". Move to the description.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: From the context, it's not clear whether the Ingress API object and the Ingress resource are two different entities or the same entity.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:335]:
```suggestion
[[ingress-adj]]
```
Each glossary entry needs to have a unique ID to ensure that anchor links work correctly.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:333]:
```suggestion
*See also*: xref:ingress-adj[ingress]
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:343]:
```suggestion
*See also*: xref:ingress[Ingress]
```

**Review by mportman12 — CHANGES_REQUESTED** [review]
Left some suggestions and comments.

**Comment by mportman12** [comment]
@skrthomas Please review suggested changes when you have a chance.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (bergerhoffer) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: @skrthomas Sorry for the late reply. The markup around `Ingress` makes sense to me. But the entry for capitalized "Ingress" still doesn't.

Per pretty much every other OCP object we have, we have the actual API object marked up when referencedd (`Pod`), but then when we speak of it generically, it's lowercase (pod). So this uppercase "Ingress" entry doesn't make sense to me. It doesn't seem like there should be a reason that "Ingress" is used but not marked up.

Or... is that the problem? Is there really no legitimate use of when "ingress"/"Ingress" should be used as a generic reference? Should it either be the `Ingress` API object (marked up), or otherwise it needs to be "ingress" as an adjective, in front of a noun?

Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: @bergerhoffer 
> It doesn't seem like there should be a reason that "Ingress" is used but not marked up.

The reason ingress could be used but not marked up would be when it is used to describe the direction of traffic flow, which is why I was thinking it could be its own entry in the Glossary. But I guess, further thinking on this, I'm wondering if we can keep a single "ingress" definition, and clarify that the Ingress resource exposes HTTP and HTTPS routes from outside the cluster to _services within the cluster_, basically routing incoming traffic to different cluster applications. 

Then we can make it clearer that you can refer to the `Ingress` object, which must be marked up, or you can describe the general directional flow of incoming traffic, which does not need to be marked up and can be lowercase. 

> Is there really no legitimate use of when "ingress"/"Ingress" should be used as a generic reference? Should it either be the Ingress API object (marked up), or otherwise it needs to be "ingress" as an adjective, in front of a noun?

In my experience writing about Networking, I've only ever used it to either reference the `Ingress Controller` resource or to describe directional flow of traffic. Maybe these are the two use cases similar to `Pod` and pod.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: I changed this to Ingress without markup when referencing the `Ingress` resource or `IngressController`

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: @mportman12 Ingress API object and Ingress resource are the same thing, but I don't think interchanging them in this definition does any good. Probably best to just stick with Ingress object. I also think more can be done in this definition to draw the connection between the API object, the Ingress Controller, and the Ingress Operator while we're at giving this a rework. I used the context here to help clarify this definition: https://docs.openshift.com/container-platform/4.12/networking/ingress-operator.html#nw-installation-ingress-config-asset_configuring-ingress

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: I was thinking actually maybe the meaning I'm trying to convey can actually be wrapped into one ingress entry, not creating a separate entry for the adjective use.

**Comment by skrthomas** [comment]
@bergerhoffer @mportman12 Apologies, I had a GA last Thursday and I've been bogged by that for a good month and a half and haven't had a chance to get back to this until now. I made some adjustments to my PR based on your feedback and questions (THANK YOU!!). I was thinking, maybe the ingress (noun) definition just needs a little more clarification and added detail to encapsulate the adjective version of the word I was trying to define here. I added more clarifying details to the existing definition in lieu of a new ingress adjective entry.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (mportman12) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]:
```suggestion
*Description*: In Red Hat OpenShift, _Ingress_  is an API object that developers can use to expose services within the cluster through an HTTP(S) aware load balancer and a proxy layer using a public DNS entry. The `Ingress` resource defines the cluster-wide configuration for ingress traffic, and provides the ability to specify TLS options, a certificate, or a public CNAME that the OpenShift Ingress Controller (`IngressController`) can accept for HTTP(S) traffic. Furthermore, _ingress_ also describes the incoming direction of network traffic. In Red Hat OpenShift, this traffic is described as entering (ingress) or leaving (egress) an OCP cluster.
```
The description is very long and detailed - not sure it needs to be so detailed.

**Review by mportman12 — CHANGES_REQUESTED** [review]

**Comment by skrthomas** [comment]
@mportman12 @bergerhoffer how do these updates look now?

**Review by mportman12 — DISMISSED** [review]
LGTM

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (bredamc) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:326]:
```suggestion
==== image:images/yes.png[yes] ingress (noun, adjective)
```

```suggestion
==== image:images/yes.png[yes] ingress
```

It seems odd to me to have two sets of parentheses. I suggest either putting both parts of speech inside a single set of parentheses, or omit the parts of speech altogether. Or split them into two entries, one for noun and one for adjective.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (bredamc) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]:
```suggestion
*Description*: In Red Hat OpenShift, _Ingress_  is an API object that developers can use to expose services within the cluster through an HTTP(S) aware load balancer and a proxy layer using a public DNS entry. The `Ingress` resource defines the cluster-wide configuration for ingress traffic, and provides the ability to specify TLS options, a certificate, or a public CNAME that the OpenShift Ingress Controller (`IngressController`) can accept for HTTP(S) traffic. Additionally, _ingress_ can also be used to describe the incoming direction of network traffic. In Red Hat OpenShift, for example, this traffic is described as entering (ingress) or leaving (egress) an OCP cluster.
```

1. Suggest putting each sentence on a new line to make it easier for reviewers to add comments :)
2. Is the first sentence necessary? We should avoid using "Ingress" on its own. 
3. What noun should be used after `IngressController`? Is it a resource or an object or something else?
4. "OCP" is not an approved abbreviation, do not use.
5. Should we add an entry for "egress", for completeness?

**Review by bredamc — COMMENTED** [review]
@skrthomas A few suggestions and questions for you to consider :)

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:326]: Thanks @bredamc , sorry for being so delayed getting back to this! Initially, I created a separate adjective entry, but initial reviews found that more confusing. So I added the adjective ingress definition to the noun ingress definition. I think I'll just leave it as is, and add the comma between the parts of speech.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: 1. ACK, good to know. Since this has already gone through most of the reviews its going to have, I think I'll forego and just make sure to do this for future style guide contributions :) Thanks for the tip though!
2. The first sentence is necessary, and I think it should be clarified to say that "_ingress_ is a Kubernetes API object". I was just following the convention of italicizing the term being defined, although you are correct that technically this usage of Ingress object would typically get backticks. 
3. resource. added!
4. ACK. Changed to OpenShift. 
5. That's a good idea. I would prefer to handle that addition in a separate PR, just so I can go ahead and merge this one. I opened it in July 😓

**Comment by skrthomas** [comment]
@bredamc thanks for your review, and apologies for the delay here. I had some other feature work and bugs take my attention but I think I've addressed your feedback. I can handle the egress addition in a separate PR entry.

@bergerhoffer @mportman12 I think this needs re-reviewed by you since I pushed my updates for Breda's feedback. TY!! I hate that this has been open since July 😓 ...

**Review by mportman12 — APPROVED** [review]
LGTM

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (dfitzmau) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: Thanks, @skrthomas , for adding this item. I've queried these trms in the past as they are highlighted illogically sometimes in customer documentation. 

Just wondering about "HTTP(S) aware load balance": I understand with the paranethesis, appending "aware" with a hyphen, which would be common, might be difficult, so how about: "an HTTP(S) load balancer" to drop the potential hyphenated construct?

Expand CNAME?

"IngressController object" more precise? I see the object reference commonly adopted in OCP docs. 

The "See also" subheading needs to be escaped onto the next line.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (dfitzmau) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:332]:
```suggestion
"Ingress" without markup When referencing the `Ingress` resource or `IngressController`.
```

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (dfitzmau) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:333]: Try adding a space before "See also" to avoid it falling back on the previous line.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (IngridT1) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: @dfitzmau I had the same question about whether or not to hyphenate "HTTP(S) aware load balance". If removing "aware" works, that would solve the issue. @skrthomas wdyt?

**Review by IngridT1 — COMMENTED** [review]
LGTM if the comments are addressed.

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: I think dropping the "aware" is totally fine. Good catch :)  It was a holdover from the original definition that I didn't change but could definitely be changed without altering the meaning.

Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: >Expand CNAME?
>"IngressController object" more precise? I see the object reference commonly adopted in OCP docs.

I do not think CNAME needs expanding.
Yea, I think `Ingress` resource and `IngressController` object is precise. thanks for pointing this out!

**Comment by skrthomas** [comment]
@IngridT1 @dfitzmau I've incorporated your feedback. Thanks for your reviews! I guess it needs another approval before its good to go? 

cc @mportman12

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (IngridT1) [inline supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc:327]: Sorry, I just notice a small issue. Could you please add a word to this sentence?

> In Red Hat OpenShift, _Ingress_ is a Kubernetes API object that developers can use to expose services within the cluster through an HTTP(S) load balancer and a proxy layer using a public DNS entry.

Please add "by" before "using a public DNS entry."

**Review by joaedwar — DISMISSED** [review]
lgtm

File: supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc
Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: Addressed :)

Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: The timing is excellent as I just stumbled across CNAME. It means "Canonical Name". Maybe put this in parentheses? JUst more a case of keeping consistent with spelling out full terms on first use.

Comment (skrthomas) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: When I search the OCP docs repo, CNAME is never expanded in parentheses so I don't really know if its necessary to do that here in this definition.

Comment (dfitzmau) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: @redhat-documentation/ccs-style-council , WDYT? OUt of context or good for consistency of rules?

Comment (mportman12) [reply supplementary_style_guide/glossary_terms_conventions/general_conventions/i.adoc]: I vote out of context

**Review by dfitzmau — APPROVED** [review]

**Review by mportman12 — APPROVED** [review]
LGTM

**Pull request merged**
